swig/cpp/src/Struct.cpp

```diff
@@ -0,0 +1,539 @@
+/**
+ * @file Struct.cpp
+ * @author Mislav Novakovic <mislav.novakovic@sartura.hr>
+ * @brief Sysrepo class header implementation for C struts.
+ *
+ * @copyright
+ * Copyright 2016 Deutsche Telekom AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+#include <memory>
+
+#include "Struct.h"
+#include "Sysrepo.h"
+
+extern "C" {
+#include "sysrepo.h"
+}
+
+using namespace std;
+
+Operation::Operation(sr_change_oper_t oper) {_oper = oper;}
+Operation::~Operation() {return;}
+
+Char_val::Char_val(char *data) {_data = data;}
+Char_val::~Char_val() {return;}
+Bool_val::Bool_val(bool data) {_data = data;}
+Bool_val::~Bool_val() {return;}
+Double_val::Double_val(double data) {_data = data;}
+Double_val::~Double_val() {return;}
+Int8_val::Int8_val(int8_t data) {_data = data;}
+Int8_val::~Int8_val() {return;}
+Int16_val::Int16_val(int16_t data) {_data = data;}
+Int16_val::~Int16_val() {return;}
+Int32_val::Int32_val(int32_t data) {_data = data;}
+Int32_val::~Int32_val() {return;}
+Int64_val::Int64_val(int64_t data) {_data = data;}
+Int64_val::~Int64_val() {return;}
+Uint8_val::Uint8_val(int8_t data) {_data = data;}
+Uint8_val::~Uint8_val() {return;}
+Uint16_val::Uint16_val(int16_t data) {_data = data;}
+Uint16_val::~Uint16_val() {return;}
+Uint32_val::Uint32_val(int32_t data) {_data = data;}
+Uint32_val::~Uint32_val() {return;}
+Uint64_val::Uint64_val(int64_t data) {_data = data;}
+Uint64_val::~Uint64_val() {return;}
+
+// Data
+Data::Data(sr_data_t data, sr_type_t type) {_d = data; _t = type;}
+Data::~Data() {return;}
+shared_ptr<Char_val> Data::get_binary() {
```

Comment:
I am not sure if we misunderstood each other here or this was necessary because of memory control. I don't see benefit in having a class for each data type.  
I expected that:
- if the type matches (value type is `SR_BINARY_T` and `get_binary` method is used) that "primitive" value is returned. IMHO there is redundant `->get()` currently.
- if the incorrect type is requested an exception is thrown

However, if it is necessary for some reason or you find it more suitable I am OK with that.